lib/src/rules/avoid_types_on_closure_parameters.dart

```diff
@@ -12,6 +12,10 @@ const _details = r'''
 
 **AVOID** annotating types on function expressions.
 
+Annotating types in function expressions is usually unnecessary because
+function expressions are almost always used on multiple values of the same
+type, thus making the practice redundant.
+
```

Comment:
Consider adding something like: Usually the type should be inferred from the context, e.g. the function is provided as a callback argument.